src/Microsoft.AspNetCore.Razor.Evolution/Intermediate/ChecksumIRNode.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Security.Cryptography;
+using System.Text;
+using Microsoft.AspNetCore.Razor.Evolution.Legacy;
+
+namespace Microsoft.AspNetCore.Razor.Evolution.Intermediate
+{
+    public class ChecksumIRNode : RazorIRNode
+    {
+        public override IList<RazorIRNode> Children => EmptyArray;
+
+        public override RazorIRNode Parent { get; set; }
+
+        internal override MappingLocation SourceRange { get; set; }
+
+        public string Bytes { get; set; }
+
+        public string Filename { get; set; }
+
+        public string Guid { get; set; }
+
+        public override void Accept(RazorIRNodeVisitor visitor)
+        {
+            visitor.VisitChecksum(this);
+        }
+
+        public override TResult Accept<TResult>(RazorIRNodeVisitor<TResult> visitor)
+        {
+            return visitor.VisitChecksum(this);
+        }
+
+        public static ChecksumIRNode Create(RazorSourceDocument sourceDocument)
```

Comment:
Instead of having this on the Checksum node we could have this as part of the source document. This felt cleaner though.